src/Collapsable.jsx

```diff
@@ -0,0 +1,99 @@
+var React = require('react');
+var joinClasses = require('./utils/joinClasses');
+var BootstrapMixin = require('./BootstrapMixin');
+var CollapsableMixin = require('./CollapsableMixin');
+var classSet = require('./utils/classSet');
+var domUtils = require('./utils/domUtils');
+var cloneWithProps = require('./utils/cloneWithProps');
+
+var ValidComponentChildren = require('./utils/ValidComponentChildren');
+var createChainedFunction = require('./utils/createChainedFunction');
+
+
+var Collapsable = React.createClass({
+  mixins: [BootstrapMixin, CollapsableMixin],
+
+  propTypes: {
+    onSelect: React.PropTypes.func,
+    expanded: React.PropTypes.bool,
+    eventKey: React.PropTypes.any
+  },
+
+  getCollapsableDOMNode: function () {
+    return this.getDOMNode();
+  },
+
+  getCollapsableDimensionValue: function () {
+    var height = 0;
+    var nodes = this.refs;
+    for (var key in nodes) {
+      if (nodes.hasOwnProperty(key)) {
+        var n = nodes[key].getDOMNode(), h = n.offsetHeight, computedStyles = domUtils.getComputedStyles(n);
+        height += (h + parseInt(computedStyles.marginTop, 10) + parseInt(computedStyles.marginBottom, 10));
+      }
+    }
+    return height;
+  },
+
+  render: function () {
+
+    var classes = this.props.collapsable ? this.getCollapsableClassSet() : {};
```

Comment:
Note that [`this.getCollapsableClassSet()`](https://github.com/react-bootstrap/react-bootstrap/blob/v0.15.1/src/CollapsableMixin.js#L101-L117) Will break apart classes passed in via props as well, so can you change the false case to be the `classSet` representation of `this.props.className`?